### DISCUSSION TITLE
Scripting simulation in Matlab help

### DISCUSSION THREAD

**Discussion by DarthTroopers** [simnibs/simnibs#331] [6. Getting started] [answered]
Hello,

I am brand new to SimNIBS and have very little Matlab experience. I got through the "Setting up and Running Simulations" tutorial, however, I am having a lot of difficulty adding the SimNIBs functions to MATLAB. I added the SimNIBS installer folder to the Matlab path.
![image](https://github.com/simnibs/simnibs/assets/169263846/64382b1e-f259-4f4e-9412-c11690441681)

However, I do not know how to access the default installations to call the addpath function. When I enter the following code

addpath('C:\Users\<USER_NAME>\AppData\Local\SimNIBS\matlab_tools')

% Initialize a session
s = sim_struct('SESSION');
% Name of head mesh
s.subpath = 'm2m_ernie';
% Output folder
s.pathfem = 'tutorial/';

into the command window, I get an error that the functions are not defined. I believe this results from the SimNIBs functions not being properly added into Matlab, however, I have no idea how to fix this. Any help is appreciated!

**Comment by oulap** [comment, marked as answer]
Hi,

You should edit the addpath call to point to where SimNIBS is installed. It seems that your username is fence. You can first try to open a file browser and navigate to this location:

C:\Users\fence\AppData\Local\

You should see a folder called SimNIBS or SimNIBS-4.1 there. If that's the case please replace the standard addpath call with this:

addpath('C:\Users\fence\AppData\Local\SimNIBS\matlab_tools')

and try running again. If you do not see a SimNIBS folder inside that AppData\Local-folder then the either the installation failed or it got installed in another location.

**Comment by ostowB** [comment]
Hello, 

I am also new to MATLAB and am trying to run scripts, analysis, and optimization with simNIBS. I was wondering if I needed to purchase any of the additional toolboxes in order to run this code.

**Reply by oulap** [reply]
The matlab code calls python under the hood, so there shouldn't be a need for any specific matlab toolboxes.
